proof/invariant-abstract/CSpaceInvPre_AI.thy

```diff
@@ -141,8 +136,9 @@ lemma empty_table_caps_of:
   "empty_table S ko \<Longrightarrow> caps_of ko = {}"
   by (cases ko, simp_all add: empty_table_def caps_of_def cap_of_def)
 
-context begin interpretation Arch .
-lemma free_index_update_test_function_stuff[simp]:
+(* FIXME arch_split: exports properties of functions that are not necessarily in global context,
+   and then they get placed in the global simpset *)
+lemma (in Arch) free_index_update_test_function_stuff[simp]:
```

Comment:
This is the slightly stranger case. We probably should avoid this kind of Arch lemma inside generic files. Then again, it has the same kind of remove-duplication benefit as the `crunch (bcorres)` we had elsewhere, so maybe we don't need to be too strict.

Thumbs up to making the name available so it can be removed from `[simp]` again if needed.

Reply:
I think this is a result of the early approaches and "Pre" tech, I think we would try to avoid this in new files.